### DISCUSSION TITLE
Exporting slides titles from Powerpoint to Freeplane

### DISCUSSION THREAD

**Discussion by EdoFro** [freeplane/freeplane#1082] [Show and tell]
Hi,
Here is a little vba macro to be used in powerpoint, that puts in the clipboard the structure of the presentation: its sections and their slides.

You can run this macro and then open Freeplane and paste the structure as a new branch.

I use this to use Freeplane to (re)organize a presentation, adding and controlling pending tasks and pending information.

Here it is:
```vba
Sub listSectionsAndSlides()
    'as outline for Freeplane
    Dim lamina As Slide
    Dim texto As String
    Dim titulo As String
    Dim ppt As Presentation
    Dim previousSectionIndex  As Integer
    Dim withTitle As Boolean
    
    Set ppt = ActivePresentation
    
    texto = ""
    previousSectionIndex = 0
    
    For Each lamina In ppt.Slides
        With lamina
            If .sectionIndex <> previousSectionIndex And .sectionIndex <= ppt.SectionProperties.Count Then
                While previousSectionIndex < .sectionIndex
                    previousSectionIndex = previousSectionIndex + 1
                    texto = texto & ppt.SectionProperties.Name(previousSectionIndex) & vbNewLine
                Wend
            End If
            withTitle = .Shapes.HasTitle
            titulo = ""
            If withTitle Then
                titulo = .Shapes.Title.TextFrame2.TextRange.Text
            End If
            If titulo = "" Then
                titulo = "id_" & .SlideID & "  - " & .Name
            End If
            texto = texto & vbTab & titulo & vbNewLine
        End With
    Next lamina
    Clipboard texto
    Beep
End Sub

```
Edited: to also list empty sections

This uses a "clipboard" function, that I found [in thespreadsheetguru](https://www.thespreadsheetguru.com/blog/2015/1/13/how-to-use-vba-code-to-copy-text-to-the-clipboard), originaly posted in [StackOverflow forum](https://stackoverflow.com/questions/14219455/excel-vba-code-to-copy-a-specific-string-to-clipboard/60896244#60896244) by [ExcelHero](https://www.excelhero.com/).

```vba
Function Clipboard(Optional StoreText As String) As String
    'PURPOSE: Read/Write to Clipboard
    'Source: ExcelHero.com (Daniel Ferry)
    
    Dim x As Variant
    Dim myClipboardData As Object
    
    'Store as variant for 64-bit VBA support
    x = StoreText
    
    'Create HTMLFile Object
    With CreateObject("htmlfile")
        Set myClipboardData = .parentWindow.clipboardData
        With myClipboardData
            Select Case True
                Case Len(StoreText)
                    'Write to the clipboard
                    .setData "text", x
                Case Else
                    'Read from the clipboard (no variable passed through)
                    On Error Resume Next
                    Clipboard = .GetData("text")
                    .clearData "Text"
            End Select
        End With
    End With
End Function
```

### how to use it
1. add a new macro module to a powerpoint presentation (in vba editor). 
2. paste both codes into the module
3. activate the powerpoint window that has the presentation you want
4. run **listSectionsAndSlides** sub
5. paste in freeplane

Hope it helps,

edo

**Comment by Illegit** [comment]
Hi Edo, thank you for sharing but in my case, it is working only for the titles of the slides, but not for any content. Just to inform you.

**Reply by EdoFro** [reply]
It works that way. It just makes an outline of sections and slide titles. 

I work my slides directly in powerpoint (sometimes using the outline view to resemble Freeplane).

I use the macro above to paste the slides titles and put in those nodes which are the pending tasks related to each slide. I use it when revisiting a presentation that I have to update for a new meeting. 

For example, now I'm working on a ppt that I presented in December. I have to update it so it reflects the new state of the art.

bye,
edo

**Reply by Illegit** [reply]
Ok, clear, therefore it is working correctly!
